clients/pkg/promtail/targets/docker/target_group.go

```diff
@@ -60,11 +66,36 @@ func (tg *targetGroup) sync(groups []*targetgroup.Group) {
 // addTarget checks whether the container with given id is already known. If not it's added to the this group
 func (tg *targetGroup) addTarget(id string, discoveredLabels model.LabelSet) error {
 	if tg.client == nil {
-		var err error
+		hostURL, err := url.Parse(tg.host)
+		if err != nil {
+			return err
+		}
+
 		opts := []client.Opt{
 			client.WithHost(tg.host),
 			client.WithAPIVersionNegotiation(),
 		}
+
+		// There are other protocols than HTTP supported by the Docker daemon, like
+		// unix, which are not supported by the HTTP client. Passing HTTP client
+		// options to the Docker client makes those non-HTTP requests fail.
+		if hostURL.Scheme == "http" || hostURL.Scheme == "https" {
+			rt, err := config.NewRoundTripperFromConfig(tg.httpClientConfig, "docker_sd")
+			if err != nil {
+				return err
+			}
+			opts = append(opts,
+				client.WithHTTPClient(&http.Client{
+					Transport: rt,
+					Timeout:   0, // Use 0 timeout, means it it reads until all has been read.
```

Comment:
I think this should be configurable within the target group config.

See Prometheus implementation as an example:
https://github.com/prometheus/prometheus/blob/main/discovery/moby/docker.go#L153